react/logic/parseDateTimeFields.ts

```diff
@@ -0,0 +1,17 @@
+interface ParseDateTimeParams {
+  data: Record<string, any>
+  properties: Record<string, unknown>
+}
+
+export const parseDateTimeFieldsData = ({
+  data,
+  properties,
+}: ParseDateTimeParams) => {
+  Object.entries(properties).forEach(([propertyKey, property]) => {
+    if ((property as any)?.format === 'date-time' && data[propertyKey]) {
+      data[propertyKey] = new Date(data[propertyKey]).toISOString()
```

Comment:
Same here. Could you create a new object at the beginning of the function like `const parsedData = { ...data }` and work with that instead of directly writing on the data?